idesolver/idesolver.py

```diff
@@ -332,13 +341,20 @@ def _solve_rhs_with_known_y(self, y: np.ndarray) -> np.ndarray:
         interpolated_y = self._interpolate_y(y)
 
         def integral(x):
-            result, *_ = self.integrator(
-                lambda s: self.k(x, s) * self.f(interpolated_y(s)),
-                self.lower_bound(x),
-                self.upper_bound(x),
-                epsabs=self.int_atol,
-                epsrel=self.int_rtol,
-            )
+            def integrand(s):
+                return np.array(self.k(x, s) * self.f(interpolated_y(s)),
```

Comment:
I did this in an external function (since not attribute is needed for the `coerce_to_array` function.
Also I moved all the coercions in `__init__`, so the user can provide vectorized version of `c`, `d`, `k` and `f` with list or numpy array without having to care about it.